libs/timeseries/TimeSeriesIndicators.h

```diff
@@ -91,7 +100,9 @@ namespace mkc_timeseries
   template <class Decimal>
   NumericTimeSeries<Decimal> RocSeries (const NumericTimeSeries<Decimal>& series, uint32_t period)
   {
-    unsigned long initialEntries = std::max (series.getNumEntries() - 1, (unsigned long) 1);
+    size_t n = series.getNumEntries();
+    size_t cap = (n > period ? n - period : 0);
```

Comment:
Should this be 
```suggestion
    size_t cap = (n > period ? n - period : n);
```

Reply:
No. If the input series has fewer than period+1 points, you can’t compute the ROC at all. By forcing cap to zero in the n <= period case, we ensure no “phantom” ROC values are ever emitted. 

Using your suggestion then in the n <= period branch you’d end up pretending you have n ROC‐values to emit even though there aren’t enough historical points to compute them. You would march off the end of the input series when you try to access “value period ago”.